src/main/java/org/littleshoot/proxy/mitm/BouncyCastleSslEngineSource.java

```diff
@@ -56,8 +57,9 @@
  */
 public class BouncyCastleSslEngineSource implements SslEngineSource {
 
-    private static final Logger LOG = LoggerFactory
-            .getLogger(BouncyCastleSslEngineSource.class);
+    private static final Logger LOG = LoggerFactory.getLogger(BouncyCastleSslEngineSource.class);
+
+    public static final String SSLENGINE_IMPL_TRY_HOST_NAME_VERIFICATION = "sun.security.ssl.SSLEngineImpl#tryHostNameVerification";
```

Comment:
This code is from `tryHostNameVerificationJava6(SSLEngine)`. It tries to increase security by enabling host name verification, since it is available only with URLConnection based HTTPS clients in Java 6 and Android. This reflection based access to internal classes isn't working at all. So I'll remove the method completely. 

**Note:** LittleProxy-mitm doesn't verify if the host name is matching to the CN or the Subject Alternative Name of a certificate on JDK 6 and **Android**. This includes Android library based frameworks like **RoboVM**, **Codename One**. The Method `SSLParameters#setEndpointIdentificationAlgorithm()` is not available for `SSLSocket` or `SSLEngine` based HTTPS clients. 

Beginning with JDK 7 and higher the host name is verified with LittleProxy-mitm.